editoast/src/views/layers/mod.rs

```diff
@@ -80,17 +81,30 @@ async fn layer_view(
         return Err(LayersError::new_view_not_found(view_slug, layer).into());
     }
 
-    let tiles_url_pattern = format!(
-        "{root_url}/layers/tile/{layer_slug}/{view_slug}/{{z}}/{{x}}/{{y}}/?infra={infra}",
-        root_url = get_root_url()
-    );
+    let root_url = get_root_url();
+    let mut tiles_url_pattern = Url::parse(&root_url)?;
+    tiles_url_pattern.set_path("layers");
+
+    if let Ok(mut path) = tiles_url_pattern.path_segments_mut() {
+        path.push("tile");
+        path.push(&layer_slug);
+        path.push(&view_slug);
+        path.push("{z}");
+        path.push("{x}");
+        path.push("{y}");
```

Comment:
This won't work. It will be replaced by ` %7Bx%7D`. We should find a way to skip the special character replacement.